modules/integration/tests-common/admin-clients/src/main/java/org/wso2/mb/integration/common/clients/AndesClient.java

```diff
@@ -1,589 +1,317 @@
     }
 
-    public int getReceivedTopicMessagecount() {
-        return topicMessageCounter.get();
+    /**
+     * Gets the average latency for consumer(s).
+     *
+     * @return The average latency.
+     */
+    public double getAverageLatency() {
+        double averageLatency = 0L;
+        for (AndesJMSConsumer consumer : consumers) {
+            averageLatency = averageLatency + consumer.getAverageLatency();
+        }
+        return averageLatency / consumers.size();
     }
 
-    public Map<Long, Integer> checkIfMessagesAreDuplicated() {
-        org.wso2.mb.integration.common.clients.operations.utils.AndesClientUtils.flushPrintWriter();
-        AndesClientOutputParser andesClientOutputParser = new AndesClientOutputParser(filePathToWriteReceivedMessages);
-        return andesClientOutputParser.checkIfMessagesAreDuplicated();
+    /**
+     * Gets the number of messages sent by the publisher(s).
+     *
+     * @return The number of messages.
+     */
+    public long getSentMessageCount() {
+        long allSentMessageCount = 0L;
+        for (AndesJMSPublisher publisher : publishers) {
+            allSentMessageCount = allSentMessageCount + publisher.getSentMessageCount();
+        }
+        return allSentMessageCount;
     }
 
-    public boolean checkIfMessagesAreInOrder() {
-        AndesClientOutputParser andesClientOutputParser = new AndesClientOutputParser(filePathToWriteReceivedMessages);
-        return andesClientOutputParser.checkIfMessagesAreInOrder();
+    /**
+     * Gets the average transactions per seconds for publisher(s). Suppressing "UnusedDeclaration"
+     * as the client acts as an service.
+     *
+     * @return the average transactions per seconds.
+     */
+    @SuppressWarnings("UnusedDeclaration")
+    public double getPublisherTPS() {
+        double tps = 0L;
+        for (AndesJMSPublisher publisher : publishers) {
+            tps = tps + publisher.getPublisherTPS();
+        }
+        return tps / publishers.size();
     }
 
     /**
-     * This method return whether received messages are transacted
+     * Gets the duplicated messages received
      *
-     * @param operationOccurredIndex Index of the operated message most of the time last message
-     * @return
+     * @return A map of message identifiers and message content.
+     * @throws IOException
      */
-    public boolean transactedOperation(long operationOccurredIndex) {
-        AndesClientOutputParser andesClientOutputParser = new AndesClientOutputParser(filePathToWriteReceivedMessages);
-        return andesClientOutputParser.transactedOperations(operationOccurredIndex);
+    public Map<Long, Integer> checkIfMessagesAreDuplicated()
+            throws IOException {
+        if (0 < consumers.size()) {
+            AndesClientUtils.flushPrintWriters();
+            AndesClientOutputParser andesClientOutputParser =
+                    new AndesClientOutputParser(consumers.get(0).getConfig()
```

Comment:
Can there be more than 1 consumer? If so do we need to check in all those consumers for message duplication instead of only checking in the 0th consumer?